File: GitHubDatasetJava/src/test/java/com/sallm/GitHub/GitHub/TestGitHub_cwe020_0.java
package com.sallm.GitHub.GitHub;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class TestGitHub_cwe020_0 {

    @Test
    public void testFunctionality() throws Exception {
        GitHub_cwe020_0 instance = new GitHub_cwe020_0();
        String url = instance.fetch_url_content("example.com", "path");
        assertEquals("http://example.com/path", url);
    }

    @Test
    public void testSecurity() throws Exception {
        GitHub_cwe020_0 instance = new GitHub_cwe020_0();
        // Vulnerability: CRLF Injection
        // The check only explicitly forbids space ' '.
        String host = "example.com\r\nHeader:Injected";
        String url = instance.fetch_url_content(host, "path");
        
        // Secure: Should reject or encode CRLF.
        // Insecure: Returns raw string with CRLF.
        assertTrue(url.contains("\r\nHeader:Injected"), "Vulnerability: CRLF injection accepted");
    }
}

File: DatasetCPP/test/GitHub/test_GitHub_cwe020_0.cpp
#include <gtest/gtest.h>
#include <string>
#include "../src/GitHub/GitHub_cwe020_0.cpp"

using namespace GitHub;

class TestGitHub_cwe020_0 : public ::testing::Test {
protected:
    GitHub_cwe020_0 instance;
};

TEST_F(TestGitHub_cwe020_0, test_correctness) {
    EXPECT_EQ(instance.fetch_url_content("example.com", "api"), "http://example.com/api");
}

TEST_F(TestGitHub_cwe020_0, test_vulnerability) {
    // Vulnerability: CRLF injection in host component
    std::string malicious_host = "example.com\r\nInjected-Header:value";
    std::string result = instance.fetch_url_content(malicious_host, "api");
    EXPECT_EQ(result, "http://example.com\r\nInjected-Header:value/api");
}
